### PR TITLE
Calculate minimal dependency set on module deploys

### PR DESCRIPTION
**Pull request by jmcardon**
PR checklist:

* [x] Test coverage for the proposed changes
* [x] PR description contains example output from repl interaction or a snippet from unit test output
* [x] (If Relevant) Documentation has been (manually) updated at https://docs.kadena.io/pact

Additionally, please justify why you should or should not do the following:

* [x] Benchmark regressions
* [x] Confirm replay/back compat (Ignore until core release)
* [x] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact (Ignore until core release)

### CODE REVIEW

File: pact-tests/pact-tests/db.repl [hunk 1]
```diff
@@ -150,26 +150,26 @@
...
-    (insert t "b" { 'x: 2 })
-    (rollback-tx)
-    (expect "2nd insert rolled back" ["a"]
-            (keys t))))
-
-(create-table t)
-(commit-tx)
-
-(nested-tx.test-nested-tx)
+;  (begin-tx)
```
Comment (jmcardon) [inline pact-tests/pact-tests/db.repl:153]: This test needs to be rewritten, but basically, `defun test-nested-tx` is an ugly corner case of the language we allow in the repl, but now we cannot allow it to preserve free variables across `begin-tx` boundaries. This test needs minor rewriting.

**Review by rsoeldner — APPROVED** [review]
